src/commonMain/kotlin/exchange.dydx.abacus/calculator/TriggerOrdersInputCalculator.kt

```diff
@@ -56,102 +71,176 @@ internal class TriggerOrdersInputCalculator(val parser: ParserProtocol) {
-        modified.safeSet("side", side?.rawValue)
-        return modified
-    }
-
-    private fun getOrderSide(
-        position: Map<String, Any>,
-    ): OrderSide? {
-        val positionSide = parser.asString(parser.value(position, "resources.indicator.current"))
+        val modified = triggerOrder.mutable()
+        val triggerPrices = parser.asNativeMap(triggerOrder["price"])?.let { calculateTriggerPrices(it, position, orderSize) }
+        modified.safeSet("price", triggerPrices)
 
-        return when (positionSide) {
-            "short" -> OrderSide.buy
-            "long" -> OrderSide.sell
-            else -> null
-        }
+        return finalizeOrderFromPriceInputs(modified, position)
     }
 
-    private fun calculatePrice(
-        triggerOrder: Map<String, Any>,
+    private fun calculateTriggerPrices(
+        triggerPrices: Map<String, Any>,
         position: Map<String, Any>,
+        orderSize: Double?,
     ): MutableMap<String, Any> {
-        val modified = triggerOrder.mutable()
+        val modified = triggerPrices.mutable()
         val entryPrice = parser.asDouble(parser.value(position, "entryPrice.current"))
-        val inputType = parser.asString(parser.value(modified, "price.input"))
+        val inputType = parser.asString(parser.value(modified, "input"))
+        val positionSide = parser.asString(parser.value(position, "resources.indicator.current"))
+        val absOrderSize = orderSize?.abs() ?: return modified
 
         if (entryPrice != null) {
-            val triggerPrice = parser.asDouble(parser.value(modified, "price.triggerPrice"))
-            val usdcDiff = parser.asDouble(parser.value(modified, "price.usdcDiff"))
-            val percentDiff = parser.asDouble(parser.value(modified, "price.percentDiff"))
+            val triggerPrice = parser.asDouble(parser.value(modified, "triggerPrice"))
+            val usdcDiff = parser.asDouble(parser.value(modified, "usdcDiff"))
+            // val percentDiff = parser.asDouble(parser.value(modified, "percentDiff")) TODO: CT-694
 
             when (inputType) {
                 "stopLossOrder.price.triggerPrice" -> {
                     modified.safeSet(
-                        "price.usdcDiff",
-                        if (triggerPrice != null) entryPrice.minus(triggerPrice).abs() else null,
-                    )
-                    modified.safeSet(
-                        "price.percentDiff",
-                        if (triggerPrice != null) Numeric.double.ONE.minus(triggerPrice.div(entryPrice)) else null,
+                        "usdcDiff",
+                        if (triggerPrice != null) {
+                            when (positionSide) {
+                                "long" -> absOrderSize.times(entryPrice.minus(triggerPrice))
+                                "short" -> absOrderSize.times(triggerPrice.minus(entryPrice))
+                                else -> null
+                            }
```

Comment:
cc @mike-dydx , here are the calculations - ended up commenting out percentDiff for now (and not setting it at all) until we come up with a better formula